integration-tests/src/test/scala/tech/beshu/ror/integration/suites/base/BaseAdminApiSuite.scala

```diff
@@ -236,14 +238,199 @@ trait BaseAdminApiSuite
         }
       }
     }
+    "provide a method for get current test settings" which {
+      "return info that test settings are not configured" when {
+        "get current test settings" in {
+          val testSettingsResponse = ror1WithIndexConfigAdminActionManager.currentRorTestSettings
+          testSettingsResponse.responseCode should be(200)
+          testSettingsResponse.responseJson("status").str should be("TEST_SETTINGS_NOT_CONFIGURED")
+          testSettingsResponse.responseJson("message").str should be("ROR Test settings are not configured")
+        }
+        "get local users" in {
+          val localUsersResponse = ror1WithIndexConfigAdminActionManager.currentRorLocalUsers
+          localUsersResponse.responseCode should be(200)
+          localUsersResponse.responseJson("status").str should be("TEST_SETTINGS_NOT_CONFIGURED")
+          localUsersResponse.responseJson("message").str should be("ROR Test settings are not configured")
+        }
+      }
+      "return current configuration" when {
+        "configuration is valid and response without warnings" in {
+          def forceReload(rorSettingsResource: String) = {
+            val testSettings = getResourceContent(rorSettingsResource)
+            val result = ror1WithIndexConfigAdminActionManager.updateRorTestSettings(testSettings)
+
+            result.responseCode should be(200)
+            result.responseJson("status").str should be("OK")
+            result.responseJson("message").str should be("updated settings")
+
+            val testSettingsResponse = ror1WithIndexConfigAdminActionManager.currentRorTestSettings
+            testSettingsResponse.responseCode should be(200)
+            testSettingsResponse.responseJson("status").str should be("TEST_SETTINGS_PRESENT")
+            testSettingsResponse.responseJson("settings").str should be(testSettings)
+            testSettingsResponse.responseJson("ttl").str.matches("""^(0|[1-9][0-9]*) seconds$""") should be(true)
+            isIsoDateTime(testSettingsResponse.responseJson("valid_to").str) should be(true)
+            testSettingsResponse.responseJson("warnings").arr should be(empty)
+          }
+
+          val dev1Ror1stInstanceSearchManager = new SearchManager(
+            clients.head.basicAuthClient("admin1", "pass"),
+            Map("IMPERSONATE_AS" -> "dev1")
```

Comment:
BTW, let's change the header name to `X-ROR-Impersonating` (as agreed somewhere in ROR KBN reviews)